pkg/kubelet/cm/deviceplugin/endpoint_store.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceplugin
+
+import (
+	"sync"
+)
+
+type store interface {
+	Get(resourceName string) (*endpoint, bool)
+	Delete(resourceName string)
+
+	Replace(resourceName string, e *endpoint) (*endpoint, bool)
+	Range(f func(k string, e *endpoint))
+}
+
+type endpointStore struct {
+	mutex     sync.Mutex
+	endpoints map[string]*endpoint // Key is the ResourceName
+}
+
+func newEndpointStore() store {
+	return &endpointStore{
+		endpoints: make(map[string]*endpoint),
+	}
+}
+
+func (s *endpointStore) Get(resourceName string) (*endpoint, bool) {
```

Comment:
Think it's called `Load()` in https://github.com/golang/go/blob/master/src/sync/map.go#L101